File: Sources/BraintreeThreeDSecure/V2UICustomization/BTThreeDSecureV2ToolbarCustomization.m
#import "BTThreeDSecureV2BaseCustomization_Internal.h"

#if __has_include(<Braintree/BraintreeThreeDSecure.h>)
#import <Braintree/BTThreeDSecureV2ToolbarCustomization.h>
#else
#import <BraintreeThreeDSecure/BTThreeDSecureV2ToolbarCustomization.h>
#endif

@implementation BTThreeDSecureV2ToolbarCustomization

- (instancetype)init {
    self = [super init];
    if (self) {
        self.cardinalValue = [ToolbarCustomization new];
    }

    return self;
}

- (void)setBackgroundColor:(NSString *)backgroundColor {
    _backgroundColor = backgroundColor;
    ((ToolbarCustomization *)self.cardinalValue).backgroundColor = backgroundColor;
}

- (void)setHeaderText:(NSString *)headerText {
    _headerText = headerText;
    ((ToolbarCustomization *)self.cardinalValue).headerText = headerText;
}

- (void)setButtonText:(NSString *)buttonText {
    _buttonText = buttonText;
    ((ToolbarCustomization *)self.cardinalValue).buttonText = buttonText;
}

@end

File: Sources/BraintreeThreeDSecure/V2UICustomization/BTThreeDSecureV2ToolbarCustomization.swift
import Foundation
import CardinalMobile

/// Toolbar customization options for 3D Secure 2 flows.
@objcMembers public class BTThreeDSecureV2ToolbarCustomization: BTThreeDSecureV2BaseCustomization {

    // MARK: - Public Properties

    /// Color code in Hex format. For example, the color code can be “#999999”.
    public var backgroundColor: String? {
        get { _backgroundColor }
        @objc(setBackgroundColor:) set {
            _backgroundColor = newValue
            (cardinalValue as? ToolbarCustomization)?.backgroundColor = newValue
        }
    }

    /// Text for the header.
    public var headerText: String? {
        get { _headerText }
        @objc(setHeaderText:) set {
            _headerText = newValue
            (cardinalValue as? ToolbarCustomization)?.headerText = newValue
        }
    }

    /// Text for the button. For example, “Cancel”.
    public var buttonText: String? {
        get { _buttonText }
        @objc(setButtonText:) set {
            _buttonText = newValue
            (cardinalValue as? ToolbarCustomization)?.buttonText = newValue
        }
    }

    // MARK: - Private Properties

    /// Used as a holder for Obj-C interoperability
    private var _backgroundColor: String?

    /// Used as a holder for Obj-C interoperability
    private var _headerText: String?

    /// Used as a holder for Obj-C interoperability
    private var _buttonText: String?

    // MARK: - Initializer

    public override init() {
        super.init()
        cardinalValue = ToolbarCustomization()
    }
}
